tests/test_keras_integration.py

```diff
@@ -34,13 +38,17 @@ def repo_name(id=uuid.uuid4().hex[:6]):
 
 def require_tf(test_case):
     """
-    Decorator marking a test that requires TensorFlow.
+    Decorator marking a test that requires TensorFlow, graphviz and pydot.
 
-    These tests are skipped when TensorFlow isn't installed.
+    These tests are skipped when TensorFlow, graphviz and pydot are installed.
 
     """
-    if not is_tf_available():
-        return unittest.skip("test requires Tensorflow")(test_case)
+    if (
+        not is_tf_available()
+        and not is_pydot_available()
```

Comment:
So TF tests won't run if pydot and graphviz aren't available? Shouldn't they still run if those two packages aren't available? I guess just what you're adding here doesn't need to be tested if those two aren't there.

Reply:
All of the TF tests are training a model and pushing to hub anyway and in the process they're plotting in the model card. So I first thought I'd split tests into two, one with TF and one with TF + plotting stuff and I thought it would be ok to have one dep checker to rule them all.